qualtran/surface_code/quantum_error_correction_scheme.py

```diff
@@ -0,0 +1,72 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import attr
+import numpy as np
+from attrs import frozen
+
+from qualtran.surface_code.physical_parameters import PhysicalParameters
+
+
+@frozen
+class QuantumErrorCorrectionScheme:
+    error_rate_scaler = attr.ib(type=float, default=0.1, repr=lambda x: f'{x:g}')
+    error_rate_threshold = attr.ib(type=float, default=0.01, repr=lambda x: f'{x:g}')
+    reference = attr.ib(type=str, default='')
+
+    def logical_error_rate(
```

Comment:
there's a function elsewhere that does this. If you intend to replace it, please remove the existing function and update usages